lib/Onyx.js

```diff
@@ -1525,6 +1619,21 @@ function init({
     shouldSyncMultipleInstances = Boolean(global.localStorage),
     debugSetState = false,
 } = {}) {
+    ActiveClientManager.init();
+
+    ActiveClientManager.isReady().then(() => {
+        if (!ActiveClientManager.isClientTheLeader()) {
+            return;
+        }
+        subscribeToEvents();
+    });
+
+    // If the active client changes an the current client is the leader,
+    // subscribes to the events
+    ActiveClientManager.subscribeToClientChange(() => {
+        subscribeToEvents();
```

Comment:
I'm confused why we would need this, and concerned that we're adding subscriptions without every cleaning them up

Reply:
I think we don't need it at all. Every time we opened or closed a tab (yes, it pushed a callback when a tab was closed instead of removing it)  it added the same [callback](https://github.com/Expensify/react-native-onyx/blob/8f4b3b0554b8955db7783001c1a72f3275e9b584/lib/Onyx.js#L1559-L1585) to the subscriptions array. And every time a message was received it run all the callbacks in the array were called. For non-leader tabs it returned early so it wasn't a problem but if there was i.e. 8 callbacks in the subscriptions array it would be called 8 times. For a new tab aka the leader tab we have always 2 different subscriptions and it's ok.

I removed this part of code and it works fine. For non leader tabs we also have only two callbacks in the subscriptions array. So I think now we don't need to add the unsubscribe logic here anymore.